packages/components/src/components/files/WebDav.ts

```diff
@@ -65,7 +75,8 @@ export function getWebDavFiles(
     containerPath: string,
     directory?: string,
     includeDirectories?: boolean,
-    skipAtFiles?: boolean
+    skipAtFiles?: boolean,
+    onlyPermittedDirectories?: boolean // Will return only uploadable directories
```

Comment:
This is also rather specific to the moduleEditor scenario. I'd suggest changing this to an optional filterCondition param. Then the implementation can specify the filter condition to use and at line 90 below it can be optionally used if it is not undefined.

Reply:
I agree this is a bit too moduleEditor-specific!
Quick question, currently the filtering condition used to find directories relies on the variable created in the reduce(), `file`. How could we pass in a condition as a function parameter, that is able to produce the same effect?

Reply:
Sure. I was thinking it would look something like this. The function you pass in would be defined as:
`(file) => file.collection && file.canUpload` 
and then inside of this usage it could be something like:
`const filterCondition = alternateFilterCondition ? alternateFilterCondition(file) : (includeDirectories || !file.collection);`
NOTE: I haven't tested this to see if it actually works or not.

Reply:
Oh, very cool, I didn't know that scoping allowed this! It works.